dropshot/tests/fail/unused_endpoint.stderr

```diff
@@ -0,0 +1,11 @@
+error: constant is never used: `unused_endpoint`
```

Comment:
Since the behavior is "constant is never used", instead of "endpoint not registered", this presumably wouldn't fire with a `pub` symbol, right?

(I think that's probably okay, we can't catch everything, but just checking my understanding)

Reply:
Yeah. Good catch. If the function is `pub` then I'm not sure there's much we can do. This is unfortunate as one is more likely to lose track of an endpoint in a larger program and one is also more likely to have endpoints in several files in larger programs.

Perhaps in the future we could consider adding endpoints to a distributed slice (https://github.com/dtolnay/linkme) to avoid manual registration entirely.

At least this change helps in simple cases as folks ramp up on Dropshot...